Sources/TokamakLint/TokenVisitor.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  TokenVisitor.swift
+//  SwiftSyntax
+//
+//  Created by Matvii Hodovaniuk on 4/2/19.
+//
+
+import Foundation
+import SwiftSyntax
+
+class TokenVisitor: SyntaxVisitor {
+  public var tree = [Node]()
+  public var current: Node?
+
+  var row = 0
+  var column = 0
+
+  override func visitPre(_ node: Syntax) {
+    var syntax = "\(type(of: node))"
+    if syntax.hasSuffix("Syntax") {
+      syntax = String(syntax.dropLast(6))
+    }
+
+    let node = Node(text: syntax)
+    node.range.startRow = row
+    node.range.startColumn = column
+    node.range.endRow = row
+    node.range.endColumn = column
+    if let current = current {
+      current.add(node: node)
+    } else {
+      tree.append(node)
+    }
+    current = node
+  }
+
+  override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
+    guard let current = current else { return .visitChildren }
+    current.text = escapeHtmlSpecialCharacters(token.text)
+    current.token = Node.Token(
+      kind: "\(token.tokenKind)"
+    )
+
+    current.range.startRow = row
+    current.range.startColumn = column
+
+    processToken(token)
+
+    current.range.endRow = row
+    current.range.endColumn = column
+
+    return .visitChildren
+  }
+
+  override func visitPost(_ node: Syntax) {
+    current?.range.endRow = row
+    current?.range.endColumn = column
+    current = current?.parent
+  }
+
+  private func processToken(_ token: TokenSyntax) {
+    var kind = "\(token.tokenKind)"
+    if let index = kind.firstIndex(of: "(") {
+      kind = String(kind.prefix(upTo: index))
+    }
+    if kind.hasSuffix("Keyword") {
+      kind = "keyword"
+    }
+
+    column += token.text.count
+  }
+
+  private func escapeHtmlSpecialCharacters(_ string: String) -> String {
```

Comment:
please add a code comment here explaining why this function is needed and how it's used